### PR TITLE
ci: automate npm release

### PR DESCRIPTION
**Pull request by dpilch**
It is a little difficult to fully test this without publishing to NPM. We could make a fake release to fully test it out if we want to. I did confirm that `lerna publish from-git -y` will not publish if the HEAD commit does not have a tag so there is limited risk of accidentaly publishing.

### CODE REVIEW

**Comment by alharris-at** [comment]
How is this intended to interact w/ our release process? This would need to run after that workflow, right?

**Comment by alharris-at** [comment]
Probably want to update docs as a part of this as well.

**Comment by dpilch** [comment]
Oh yeah, I'll update the docs.

This would be the normal flow of create the release PR and then merging to main. After the GitHub release is created the workflow would then automatically publish to NPM.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -33,3 +33,22 @@ jobs:
           body_path: release_description.md
           draft: false
           prerelease: false
+
+  npm-release:
+    needs: github-release
```
Comment (alharris-at) [inline .github/workflows/release.yml:38]: Ahh, this line is what I'd overlooked. Didn't realize you'd already built in the workflow serialization. Nice!

**Review by alharris-at — APPROVED** [review]

**Pull request merged**